subsys/net/ip/net_context.c

```diff
@@ -763,15 +763,25 @@ int net_context_bind(struct net_context *context, const struct sockaddr *addr,
 				return -EINVAL;
 			}
 
+			struct ieee802154_context *ieee802154_context =
```

Comment:
No, no l2 stuff into net_context.c
You'll have to find another way.

Reply:
This led me to better investigate the locking strategy of the ll address in net_if. I'm now relying on net_if's own lock which is much better. No need to do any L2-specific locking to update the ll-address any more. Also reminded me to cover the corresponding ieee802154_mgmt API in the test which again revealed a bug. Thanks for this useful remark.